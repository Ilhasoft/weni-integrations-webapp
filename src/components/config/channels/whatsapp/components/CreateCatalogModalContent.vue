<template>
  <div class="modal">
    <div slot="message" class="modal__content">
      <span class="modal__content__title">{{ $t('whatsapp.create_catalog.title') }}</span>

      <div class="modal__content__header">
        <span class="modal__content__header__title">{{
          $t('whatsapp.create_catalog.header.title')
        }}</span>
        <span class="modal__content__header__description">{{
          $t('whatsapp.create_catalog.header.description')
        }}</span>
      </div>

      <div class="modal__content__form">
        <div class="modal__content__form__options">
          <div
            ref="vtexOption"
            :class="['modal__content__form__options__card', type === types.VTEX ? 'selected' : '']"
            @click="type = types.VTEX"
          >
            <div class="modal__content__form__options__card__icon">
              <img
                class="modal__content__form__options__card__icon__image"
                src="~@/assets/vtex_small.png"
              />
            </div>

            <div class="modal__content__form__options__card__content">
              <span class="modal__content__form__options__card__content__title">{{
                $t('whatsapp.create_catalog.option.vtex.title')
              }}</span>
              <span class="modal__content__form__options__card__content__description">
                {{ $t('whatsapp.create_catalog.option.vtex.description') }}
              </span>
            </div>
          </div>

          <div
            ref="metaOption"
            :class="['modal__content__form__options__card', type === types.META ? 'selected' : '']"
            @click="type = types.META"
          >
            <div class="modal__content__form__options__card__icon">
              <img
                class="modal__content__form__options__card__icon__image"
                src="~@/assets/meta_small.png"
              />
            </div>

            <div class="modal__content__form__options__card__content">
              <span class="modal__content__form__options__card__content__title">
                {{ $t('whatsapp.create_catalog.option.meta.title') }}
                <unnnic-icon
                  icon="open_in_new"
                  class="modal__content__form__options__card__content__title__icon"
                />
              </span>
              <span class="modal__content__form__options__card__content__description">
                {{ $t('whatsapp.create_catalog.option.meta.description') }}
              </span>
            </div>
          </div>
        </div>

        <span
          v-html="$t('whatsapp.create_catalog.footer')"
          class="modal__content__form__footer"
        ></span>
      </div>
    </div>
    <div class="modal__buttons">
      <unnnic-button slot="options" ref="closeButton" type="tertiary" @click="closeModal">
        {{ $t('general.Cancel') }}
      </unnnic-button>
      <!-- TODO: Check if VTEX app is enabled to enable this button -->
      <unnnic-button slot="options" ref="createButton" @click="createCatalog">
        {{ $t('general.continue') }}
      </unnnic-button>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'CreateCatalogModalContent',
    data() {
      return {
        types: {
          VTEX: 'vtex',
          META: 'meta',
        },
        type: 'vtex',
      };
    },
    methods: {
      createCatalog() {
        this.$emit('createCatalog', this.type);
      },
      closeModal() {
        this.$emit('closeModal');
      },
    },
  };
</script>

<style lang="scss" scoped>
  .modal {
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-md;

    &__buttons {
      display: flex;
      gap: $unnnic-spacing-lg;
      widows: 100%;
      flex: 1;

      ::v-deep .unnnic-button {
        width: 100%;
      }
    }

    &__content {
      display: flex;
      flex-direction: column;
      gap: $unnnic-spacing-sm;
      text-align: left;

      &__title {
        color: $unnnic-color-neutral-darkest;
        font-size: $unnnic-font-size-title-sm;
        font-weight: $unnnic-font-weight-black;
        line-height: $unnnic-font-size-title-sm + $unnnic-line-height-medium;
      }

      &__header {
        display: flex;
        flex-direction: column;
        gap: $unnnic-spacing-nano;

        &__title {
          color: $unnnic-color-neutral-darkest;
          font-size: $unnnic-font-size-body-lg;
          font-weight: $unnnic-font-weight-black;
          line-height: $unnnic-font-size-body-lg + $unnnic-line-height-medium;
        }

        &__description {
          color: $unnnic-color-neutral-cloudy;
          font-size: $unnnic-font-size-body-md;
          line-height: $unnnic-font-size-body-md + $unnnic-line-height-medium;
        }
      }

      &__form {
        display: flex;
        flex-direction: column;
        gap: $unnnic-spacing-sm;

        &__options {
          display: flex;
          flex-direction: column;
          gap: $unnnic-spacing-xs;

          &__card {
            display: flex;
            padding: $unnnic-spacing-ant $unnnic-spacing-sm;
            align-items: center;
            gap: $unnnic-spacing-sm;

            border-radius: $unnnic-border-radius-sm;
            border: $unnnic-border-width-thinner solid $unnnic-color-neutral-cleanest;
            background: $unnnic-color-neutral-lightest;

            cursor: pointer;

            &.selected {
              border-radius: $unnnic-border-radius-sm;
              border: $unnnic-border-width-thinner solid $unnnic-color-weni-600;
              background: $unnnic-color-weni-50;
            }

            &__icon {
              display: flex;
              flex-direction: column;
              justify-content: center;
              align-items: center;
              border-radius: $unnnic-border-radius-sm;
              background: rgba(59, 65, 77, 0.08);
              width: 40px;
              height: 40px;
            }

            &__content {
              display: flex;
              flex-direction: column;
              align-items: flex-start;

              &__title {
                display: flex;
                align-items: center;
                gap: $unnnic-spacing-nano;
                color: $unnnic-color-neutral-darkest;
                font-size: $unnnic-font-size-body-gt;
                font-weight: $unnnic-font-weight-bold;
                line-height: $unnnic-font-size-body-gt + $unnnic-line-height-medium;

                &__icon {
                  font-size: $unnnic-font-size-title-sm !important;
                }
              }

              &__description {
                color: $unnnic-color-neutral-cloudy;
                font-size: $unnnic-font-size-body-md;
                line-height: $unnnic-font-size-body-md + $unnnic-line-height-medium;
              }
            }
          }
        }

        &__footer {
          margin-top: $unnnic-spacing-stack-xs;
          color: $unnnic-color-neutral-cloudy;
          font-size: $unnnic-font-size-body-gt;
          line-height: $unnnic-font-size-body-gt + $unnnic-line-height-medium;
        }
      }
    }
  }
</style>
