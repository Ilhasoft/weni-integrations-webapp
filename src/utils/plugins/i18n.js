import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
