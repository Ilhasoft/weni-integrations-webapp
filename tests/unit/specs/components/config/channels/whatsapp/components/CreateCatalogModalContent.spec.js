import { mount, createLocalVue } from '@vue/test-utils';
import CreateCatalogModalContent from '@/components/config/channels/whatsapp/components/CreateCatalogModalContent.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

const mountComponent = async () => {
  const wrapper = mount(CreateCatalogModalContent, {
    localVue,
    i18n,
  });

  return { wrapper };
};

describe('components/config/channels/whatsapp/components/CreateCatalogModalContent.vue', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', async () => {
    const { wrapper } = await mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should emit createCatalog with selected type', async () => {
    const { wrapper } = await mountComponent();

    const metaOption = wrapper.findComponent({ ref: 'metaOption' });
    await metaOption.trigger('click');

    const createButton = wrapper.findComponent({ ref: 'createButton' });
    await createButton.trigger('click');

    expect(wrapper.emitted().createCatalog).toBeTruthy();
    expect(wrapper.emitted().createCatalog[0]).toEqual(['meta']);
  });

  it('should emit closeModal on close click', async () => {
    const { wrapper } = await mountComponent();

    const closeButton = wrapper.findComponent({ ref: 'closeButton' });
    closeButton.trigger('click');

    expect(wrapper.emitted().closeModal).toBeTruthy();
  });
});
