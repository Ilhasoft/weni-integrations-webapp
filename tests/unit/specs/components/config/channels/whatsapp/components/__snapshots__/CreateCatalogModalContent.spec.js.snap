// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/config/channels/whatsapp/components/CreateCatalogModalContent.vue should be rendered properly 1`] = `
<div class="modal">
  <div slot="message" class="modal__content"><span class="modal__content__title">Create catalog</span>
    <div class="modal__content__header"><span class="modal__content__header__title">Sending method</span> <span class="modal__content__header__description">Choose the way you want to send your catalog to start selling products via Whatsapp</span></div>
    <div class="modal__content__form">
      <div class="modal__content__form__options">
        <div class="modal__content__form__options__card selected">
          <div class="modal__content__form__options__card__icon"><img src="~@/assets/vtex_small.png" class="modal__content__form__options__card__icon__image"></div>
          <div class="modal__content__form__options__card__content"><span class="modal__content__form__options__card__content__title">VTEX Catalog</span> <span class="modal__content__form__options__card__content__description">
              Automatically connect your catalog through integration with the VTEX app
            </span></div>
        </div>
        <div class="modal__content__form__options__card">
          <div class="modal__content__form__options__card__icon"><img src="~@/assets/meta_small.png" class="modal__content__form__options__card__icon__image"></div>
          <div class="modal__content__form__options__card__content"><span class="modal__content__form__options__card__content__title">
              Facebook Meta
              <span data-v-4ce83152="" class="modal__content__form__options__card__content__title__icon material-symbols-rounded unnnic-icon-scheme--neutral-darkest unnnic-icon-size--md"> open_in_new </span></span> <span class="modal__content__form__options__card__content__description">
              Create a catalog on Meta's Facebook, a new tab will be opened.
            </span></div>
        </div>
      </div> <span class="modal__content__form__footer">To connect your VTEX catalog you need to have integrated the VTEX app on the platform. If you haven't integrated yet, click here</span>
    </div>
  </div>
  <div class="modal__buttons"><button data-v-2d9f2cbf="" class="unnnic-button unnnic-button--size-large unnnic-button--tertiary" slot="options">
      <!---->
      <!---->
      <!----><span data-v-2d9f2cbf="" class="unnnic-button__label">
      Cancel
      </span>
      <!---->
    </button> <button data-v-2d9f2cbf="" class="unnnic-button unnnic-button--size-large unnnic-button--primary" slot="options">
      <!---->
      <!---->
      <!----><span data-v-2d9f2cbf="" class="unnnic-button__label">
      Continue
      </span>
      <!---->
    </button></div>
</div>
`;
